### PR TITLE
Migrate API from GRPC #1

### LINKED ISSUE #1: Migrate API from GRPC
1. Migration definitions from https://github.com/iotexproject/iotex-election/blob/master/explorer/idl/explorer.idl to a profobuf file
2. Implement GetMeta() based on profobuf.
3. Implement other methods based on profobuf.

### PR DESCRIPTION
**Pull request by ashishsnigam**
fixes #1 

1) Added implementation of two more methods getCandidates and getBucketsByCandidate based on protobuf

2) Corrected existing implementation after change of candidates -> delegates

### CODE REVIEW

**Comment by ashishsnigam** [comment]
fixes #3

**Comment by ashishsnigam** [comment]
Proposed new folder structure for all grpc/protobuf based apis/server/client for code maintainability and package name free of underscores as a good go practice

File: server/ranking/ranking.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+import (
+	"encoding/hex"
+	"errors"
+	"strconv"
+	"time"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/iotexproject/iotex-election/committee"
+	pb "github.com/iotexproject/iotex-election/pb/ranking"
```
Comment (raullenchai) [inline server/ranking/ranking.go]: github.com/iotexproject/... shoud be grouped together,
other github or golang imports in another group

File: server/ranking/ranking.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	}
+
+	return &pb.ChainMeta{
+		Height:          strconv.FormatUint(height, 10),
+		TotalCandidates: uint64(len(result.Delegates())),
+	}, nil
+}
+
+// GetCandidates returns a list of candidates sorted by weighted votes
+func (s *server) GetCandidates(ctx context.Context, request *pb.GetCandidatesRequest) ([]*pb.CandidateResponse, error) {
```
Comment (CoderZhi) [inline server/ranking/ranking.go]: wrong return type, should be *pb.CandidateResponse

File: server/ranking/ranking.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+		retval[i].Candidates[i].Name = hex.EncodeToString(candidate.Name())
+		retval[i].Candidates[i].PubKey = hex.EncodeToString(candidate.BeaconPubKey())
+		retval[i].Candidates[i].TotalWeightedVotes = candidate.Score().Text(10)
+	}
+
+	return retval, nil
+}
+
+// GetBucketsByCandidate returns the buckets
+func (s *server) GetBucketsByCandidate(ctx context.Context, request *pb.GetBucketsByCandidateRequest) ([]*pb.BucketResponse, error) {
```
Comment (CoderZhi) [inline server/ranking/ranking.go]: same error here, the return type should be *pb.BucketResponse

File: server/ranking/ranking.go [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+)
+
+// server is used to implement pb.RankingServer.
+type server struct {
+	savedCustomers []*pb.RankingServer
+	c              committee.Committee
+}
+
+// GetMeta returns the meta of the chain
+func (s *server) GetMeta(ctx context.Context, empty empty.Empty) (*pb.ChainMeta, error) {
```
Comment (CoderZhi) [inline server/ranking/ranking.go]: empty *empty.Empty

**Comment by ashishsnigam** [comment]
Tomorrow will try to have a working setup where I Can run unit tests to
check code. Will change code accordingly. Need help on it.

File: server/ranking/ranking.go (hunk 2 above)
Comment (ashishsnigam) [reply server/ranking/ranking.go]: corrected it. Please see and approve 
https://github.com/iotexproject/iotex-election/pull/5

**Pull request merged**